modules/core/m_mode.c

```diff
@@ -368,8 +371,23 @@ ms_bmask(struct MsgBuf *msgbuf_p, struct Client *client_p, struct Client *source
 						parv[3][0], s, forward);
 		}
 
-		if(add_id(fakesource_p, chptr, s, forward, banlist, mode_type))
+
+		if((banptr = add_id(fakesource_p, chptr, s, forward, banlist, mode_type)) != NULL)
 		{
+			if (extended)
+			{
+				bants = atol(strtok(NULL, " "));
+				who = strtok(NULL, " ");
+
+				if (who == NULL)
+				{
+					// die here, not enough params
+				}
+
+				banptr->when = bants;
+				strcpy(banptr->who, who);
```

Comment:
```suggestion
				rb_free(banptr->who);
				banptr->who = rb_strdup(who);
```

`who` in `struct Ban` is not a static size buffer, but the result of an `rb_strdup`; do the same here.